server/src/main/java/io/crate/role/JwtProperties.java

```diff
@@ -0,0 +1,123 @@
+        return new JwtProperties(in.readString(), in.readString());
+    }
+
+    @Nullable
+    public static JwtProperties fromMap(@NotNull Map<String, Object> jwtPropertiesMap) {
+        if (jwtPropertiesMap.isEmpty() == false) {
+            String iss = Maps.get(jwtPropertiesMap, "iss");
+            ensureNotNull("iss", iss);
+            String username = Maps.get(jwtPropertiesMap, "username");
+            ensureNotNull("username", username);
+            if (jwtPropertiesMap.size() > 2) {
+                throw new IllegalArgumentException(
+                    String.format(Locale.ENGLISH, "Only 'iss' and 'username' JWT properties are allowed")
+                );
+            }
+            return new JwtProperties(iss, username);
+        }
+        return null;
+
+    }
+
+    private static void ensureNotNull(String propertyName, @Nullable String value) {
+        if (value == null) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ENGLISH, "JWT must have property '%s'", propertyName)
+            );
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(iss);
+        out.writeString(username);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject("jwt")
+            .field("iss", iss)
+            .field("username", username)
+            .endObject();
+        return builder;
+    }
+
+    public static JwtProperties fromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token currentToken;
+        String iss = null;
+        String username = null;
+        while ((currentToken = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (currentToken == XContentParser.Token.FIELD_NAME) {
+                String currentFieldName = parser.currentName();
+                currentToken = parser.nextToken();
+                switch (currentFieldName) {
+                    case "iss":
+                        if (currentToken != XContentParser.Token.VALUE_STRING) {
+                            throw new ElasticsearchParseException(
+                                "failed to parse jwt, 'iss' value is not a string [{}]", currentToken);
+                        }
+                        iss = parser.text();
+                        break;
+                    case "username":
+                        if (currentToken != XContentParser.Token.VALUE_STRING) {
+                            throw new ElasticsearchParseException(
+                                "failed to parse jwt, 'username' value is not a string [{}]", currentToken);
+                        }
+                        username = parser.text();
+                        break;
+                    default:
+                        throw new ElasticsearchParseException("failed to parse jwt");
```

Comment:
Maybe add a unit test for these cases to have some coverage

Reply:
Added